libcfnet/net.c

```diff
@@ -544,12 +544,14 @@ int SocketConnect(const char *host, const char *port,
 bool TryConnect(int sd, unsigned long timeout_ms,
                 const struct sockaddr *sa, socklen_t sa_len)
 {
+    assert(sd != -1);
     assert(sa != NULL);
 
     if (sd >= FD_SETSIZE)
     {
-        Log(LOG_LEVEL_ERR, "Open connections exceed FD_SETSIZE limit of %d",
-            FD_SETSIZE);
+        Log(LOG_LEVEL_ERR,
+            "Open connections exceed FD_SETSIZE limit (%d >= %d)",
+            sd, FD_SETSIZE);
```

Comment:
@nickanderson these logs will help debugging the FD_SETSIZE issue we've hit on windows. You might want to grab a latest windows build from jenkins after this is merged, and give it to the customer - that is if he still sees the error.